BoxContentSDK/BoxContentSDK/Operations/BOXAPIOperation.m

```diff
@@ -335,18 +358,18 @@ - (void)executeSessionTask
 - (void)executeOperation
 {
     BOXLog(@"BOXAPIOperation %@ was started", self);
-    if (![self isCancelled])
-    {
-        @synchronized(self.session)
-        {
-            //Note: if sessionTask exists, we cannot change its API request
-            //make sure you recreate sessionTask with the new API request if needed
-            [self prepareAPIRequest];
-            self.accessToken = self.session.accessToken;
+    if (![self isCancelled]) {
+        if (self.sessionTask == nil) {
```

Comment:
To avoid doing that work twice, now that it might actually get done on 'prepare'